plugins/modules/ntnx_user_groups.py

```diff
@@ -0,0 +1,270 @@
+  returned: always
+  type: str
+  sample: "3.1"
+metadata:
+  description: The user_group kind metadata
+  returned: always
+  type: dict
+  sample: {
+                "categories": {},
+                "categories_mapping": {},
+                "kind": "user_group",
+                "owner_reference": {
+                    "kind": "user",
+                    "name": "admin",
+                    "uuid": "00000000-0000-0000-0000-000000000000"
+                },
+                "spec_hash": "00000000000000000000000000000000000000000000000000",
+                "spec_version": 0,
+                "uuid": "00000000-0000-0000-0000-000000000000"
+
+            }
+spec:
+  description: An intentful representation of a user_group spec
+  returned: always
+  type: dict
+  sample: {
+    "resources": {
+                        "directory_service_user_group": {
+                            "distinguished_name": "cn=sspadmins,cn=users,dc=qa,dc=nucalm,dc=io"
+                        }
+                    }
+
+  }
+status:
+  description: An intentful representation of a user_group status
+  returned: always
+  type: dict
+  sample: {
+"execution_context": {
+                    "task_uuid": [
+                        "00000000-0000-0000-0000-000000000000"
+                    ]
+                },
+                "resources": {
+                    "access_control_policy_reference_list": [],
+                    "directory_service_user_group": {
+                        "directory_service_reference": {
+                            "kind": "directory_service",
+                            "name": "qanucalm",
+                            "uuid": "00000000-0000-0000-0000-000000000000"
+                        },
+                        "distinguished_name": "cn=sspadmins,cn=users,dc=qa,dc=nucalm,dc=io"
+                    },
+                    "display_name": "sspadmins",
+                    "projects_reference_list": [],
+                    "user_group_type": "DIRECTORY_SERVICE"
+                },
+                "state": "COMPLETE"
+
+  }
+user_group_uuid:
+  description: The created user_group uuid
+  returned: always
+  type: str
+  sample: "00000000-0000-0000-0000-000000000000"
+"""
+
+from ..module_utils import utils  # noqa: E402
+from ..module_utils.base_module import BaseModule  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.prism.user_groups import UserGroups  # noqa: E402
+
+
+def get_module_spec():
+
+    mutually_exclusive = [("name", "uuid")]
+
+    entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
+
+    saml_user_group_spec = dict(
+        idp_uuid=dict(type="str", required=True),
+        group_name=dict(type="str", required=True),
+    )
+
+    module_args = dict(
+        user_group_uuid=dict(type="str"),
+        distinguished_name=dict(type="str"),
+        idp=dict(type="dict", options=saml_user_group_spec),
```

Comment:
Only IDP ? No AD ?

Reply:
for AD just need to provide distinguished_name